presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotBrokerPageSourceBase.java

```diff
@@ -110,6 +112,30 @@ private static Double parseDouble(String value)
         }
     }
 
+    protected void setValue(Type type, BlockBuilder blockBuilder, JsonNode value)
+    {
+        if (type == null || blockBuilder == null) {
```

Comment:
nit: Else if perhaps ? I also think type and blockbuilder cannot be null by the time we get to this function.

Reply:
This `BlockBuilder` is inherited from the primitive  `setValue` which was being called directly and now is being called via this one. Do you think this check can be removed since i don't see a `requiredNonNull` check in the hierarchy?

Reply:
I see, so remove the null check from the primitive setValue one ? now that it is handled here by its caller ?

Also, what does it mean if type or blockbuilder are null ? Is that a bug or perhaps it means that an output column-field has to be skipped ? I recall we do that for when creating the PQL query with some hidden column-values (like grouping without aggregation case for example in broker PQL path)

Reply:
> I see, so remove the null check from the primitive setValue one ? now that it is handled here by its caller ?

Good point, the primitive setValue is also being called directly at the moment to set the group-by field value. I am not sure if an array column can be in group-by. @fx19880617 It'd be great if you can clarify.

> Also, what does it mean if type or blockbuilder are null ? Is that a bug or perhaps it means that an output column-field has to be skipped ? I recall we do that for when creating the PQL query with some hidden column-values (like grouping without aggregation case for example in broker PQL path)

Hmm I think even for  hidden column we set types and hidden columns wouldn't have to be added to blockbuilders? since we actually do not add them to output variables? It might makes sense to remove this check.

Reply:
Hmm, Good point @dharakk ... If an array column cannot be a group by key then we might have to fail the query during pinot pql/sql generation phase :-). I feel that we should just go ahead and forbid that, it kind of does not make much sense to group by on an array (or atleast the semantics are open to interpretation).

Also, if you think that type and blockBuilder can never be null, should we put a requireNonNull instead of removing the check ? :-P ?

Reply:
^^^^ ping

Reply:
type is coming from `PinotColumnHandle` and can't be null because there is `requiredNonNull` for type there. Although for blockBuilder i think we should keep it because they are made in `PageBuilder` and also there is indeed a hidden column check where for negative columnIndex we pass in null value (Although I am not sure if we would ever encounter it).

Reply:
Also added a check to prevent and array column becoming a groupby column